### PR TITLE
Add dokka and docs dir

### PR DESCRIPTION
**Pull request by mccorby**
## Description
Add Dokka to generate documentation of the library.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: syftlib/build.gradle [hunk 1]
```diff
@@ -29,6 +30,25 @@ task jacocoTestReport(type: JacocoReport, dependsOn: ['testDebugUnitTest', 'crea
...
+    outputFormat = 'html'
+    outputDirectory = "$buildDir/html"
+    doLast {
+        ant.move file: "$outputDirectory",
+                todir: "$rootDir/docs"
+    }
+}
+
+task dokkaMarkdown(type: org.jetbrains.dokka.gradle.DokkaTask) {
+    outputFormat = 'markdown'
```
Comment (vkkhare) [inline syftlib/build.gradle:44]: Better use the gh-markdown as the format

Comment (mccorby) [reply syftlib/build.gradle:44]: I can't see this format in Dokka.

Comment (vkkhare) [reply syftlib/build.gradle:44]: https://github.com/Kotlin/dokka#output_formats

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: [ubuntu-18.04]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Generate HTML documentation
+        run: ./gradlew :syftlib:dokka
```
Comment (vkkhare) [inline .github/workflows/documentation.yml:26]: I think keeping dokka markdown is sufficient

File: syftlib/build.gradle (hunk 1 above)
Comment (mccorby) [reply syftlib/build.gradle]: It couldn't be easier to find. Thanks.

**Comment by vkkhare** [comment]
We should make the not required classes internal so that they don't come in documentation. Let's leave only the classes with which the user will interact. Like job status message, network repositories, battery status, either, they all should be internal

File: .github/workflows/documentation.yml (hunk 2 above)
Comment (vkkhare) [reply .github/workflows/documentation.yml:26]: pushing to repo is also needed here I think

**Comment by vkkhare** [comment]
Why docs/docs in directory?

**Comment by vkkhare** [comment]
I missed a few constant declarations in networking. They should have been made internal. Can you add them in this PR or should I create an issue for that?

**Comment by vkkhare** [comment]
@all-contributors add @mccorby for documentation

**Pull request merged**